Tests/Test/ResponseTest.swift

```diff
@@ -15,56 +15,96 @@ class ResponseTest: XCTestCase {
  + "<!DOCTYPE html><html><body><h1>Hello Person</h1></body></html>"
-        XCTAssertEqual(expectedResponse, response.buildResponse(serverRequest: request))
+
+        let parsedRequest = parser.parseRequest(request: request)
+
+        XCTAssertEqual(expectedResponse, response.buildResponse(request: parsedRequest))
     }
 
     func testBuildResponseWillReturnGETResponseWith2Queries() {
         let request = "GET /hello?mname=Person&lname=Doe"
         let expectedResponse = "HTTP/1.1 200 OK\r\nContent-Length: 66\r\nContent-type: text/html\r\n\r\n"
  + "<!DOCTYPE html><html><body><h1>Hello Person Doe</h1></body></html>"
-        XCTAssertEqual(expectedResponse, response.buildResponse(serverRequest: request))
+
+        let parsedRequest = parser.parseRequest(request: request)
+
+        XCTAssertEqual(expectedResponse, response.buildResponse(request: parsedRequest))
     }
 
     func testBuildResponseWillReturnGETResponseWith3Queries() {
         let request = "GET /hello?fname=Person&lname=Doe&mname=John"
         let expectedResponse = "HTTP/1.1 200 OK\r\nContent-Length: 71\r\nContent-type: text/html\r\n\r\n"
  + "<!DOCTYPE html><html><body><h1>Hello Person John Doe</h1></body></html>"
-        XCTAssertEqual(expectedResponse, response.buildResponse(serverRequest: request))
+
+        let parsedRequest = parser.parseRequest(request: request)
+
+        XCTAssertEqual(expectedResponse, response.buildResponse(request: parsedRequest))
     }
 
     func testBuildResponseWillReturnProperGETResponseWithQueryAfterAnInitialRequest() {
         let request1 = "GET /hello?fname=Person"
         let request2 = "GET /hello"
         let expectedResponse = header200 + ResponseTest.body200 
 
-        _ = response.buildResponse(serverRequest: request1)
+        let parsedRequest1 = parser.parseRequest(request: request1)
+        _ = response.buildResponse(request: parsedRequest1)
+        let parsedRequest2 = parser.parseRequest(request: request2)
+
+        XCTAssertEqual(expectedResponse, response.buildResponse(request: parsedRequest2))
+    }
+
+    func testBuildResponseWillReturn418Response() {
+        let request = "GET /coffee"
+        let expectedResponse = "HTTP/1.1 418\r\nContent-Length: 12\r\nContent-type: text/html\r\n\r\nI'm a teapot"
+
+        let parsedRequest = parser.parseRequest(request: request)
+
+        XCTAssertEqual(expectedResponse, response.buildResponse(request: parsedRequest))
+    }
+
+    func testBuildResponseWillDecodeOperators() {
+        let request = "GET /parameters?variable_1=Operators%20%3C%2C%20%3E%2C%20%3D%2C%20!%3D%3B%20%2B%2C%20-%2C%20*%2C%20%26%2C%20%40%2C%20%23%2C%20%24%2C%20%5B%2C%20%5D%3A%20%22is%20that%20all%22%3F"
+        let expectedResponse = "HTTP/1.1 200 OK\r\nContent-Length: 127\r\nContent-type: text/html\r\n\r\n<!DOCTYPE html><html><body><h1>variable_1 = Operators <, >, =, !=; +, -, *, &, @, #, $, [, ]: \"is that all\"?</h1></body></html>" 
```

Comment:
Don't have anything particular in mind, but I wonder if there is a way to clean up the `expectedResponse`s throughout this file.

Reply:
Some helper methods would do the trick. I was thinking of mentioning this when we next talked.